GestorDeEstoque/Repositories/UsuarioRepository.cs

```diff
@@ -0,0 +1,39 @@
+using GestorDeEstoque.Data;
+using GestorDeEstoque.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace GestorDeEstoque.Repositories
+{
+    public class UsuarioRepository : IUsuarioRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        public UsuarioRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Usuario> BuscarUsuarioPorNomeUsuarioAsync(string nomeUsuario)
+        {
+            var usuario = await _context.Usuarios.FirstOrDefaultAsync(u =>
+                u.NomeUsuario == nomeUsuario
+            );
+            if (usuario == null)
+            {
+                throw new Exception("Usuário não encontrado");
+            }
+            return usuario;
+        }
+
+        public async Task<Usuario> CriarUsuarioAsync(string nomeUsuario, string senha)
+        {
+            {
+                var senhaHash = BCrypt.Net.BCrypt.HashPassword(senha);
```

Comment:
Incluir  `using BCrypt.Net;` para simplificar a chamada

```suggestion
                var senhaHash = BCrypt.HashPassword(senha);
```

Reply:
Ele não aceita remover essa parte

Reply:
Mesmo incluindo o using?